src/atomate2/common/flows/anharmonicity.py

```diff
@@ -0,0 +1,276 @@
+        phonon_doc: PhononBSDOSDoc,
+        prev_dir: str | Path | None = None,
+        temperature: float = 300,
+        one_shot_approx: bool = True,
+        seed: int | None = None,
+        atom_resolved: bool = False,
+        mode_resolved: bool = False,
+        n_samples: int = 1,
+    ) -> Flow:
+        """Create an anharmonicity workflow from a phonon calculation.
+
+        Parameters
+        ----------
+        phonon_doc: PhononBSDOSDoc
+            The document to get the anharmonicity for
+        prev_dir : str or Path or None
+            A previous calculation directory to use for copying outputs.
+        temperature: float
+            The temperature for the anharmonicity calculation
+        one_shot_approx: bool
+            If true, finds the one shot approximation of sigma^A and if false,
+            finds the full sigma^A. The default is True.
+        seed: int | None
+            Seed to use for the random number generator
+            (only used if one_shot_approx == False)
+        atom_resolved: bool
+            If true, calculate the atom-resolved sigma^A. This is false by default.
+        mode_resolved: bool
+            If true, calculate the mode-resolved sigma^A. This is false by default.
+        n_samples: int
+            Number of times displaced structures are sampled.
+            Must be >= 1 and cannot be used when one_shot_approx == True.
+        """
+        if phonon_doc.has_imaginary_modes:
+            warn(
+                "The phonon model has imaginary modes, sampling maybe incorrect.",
+                stacklevel=1,
+            )
+
+        jobs = []
+        phonon_supercell_job = get_phonon_supercell(phonon_doc)
+        jobs.append(phonon_supercell_job)
+
+        phonon_supercell = phonon_supercell_job.output
+
+        displace_supercell = displace_structure(
+            phonon_supercell=phonon_supercell,
+            force_constants=phonon_doc.force_constants,
+            temp=temperature,
+            one_shot=one_shot_approx,
+            seed=seed,
+            n_samples=n_samples,
+        )
+        jobs.append(displace_supercell)
+
+        force_eval_maker = self.phonon_maker.phonon_displacement_maker
+        force_eval_maker.name = f"{force_eval_maker.name}"
+        displacement_calcs = run_displacements(
+            displacements=displace_supercell.output,
+            phonon_supercell=phonon_supercell,
+            force_eval_maker=force_eval_maker,
+            socket=self.phonon_maker.socket,
+            prev_dir_argname=self.prev_calc_dir_argname,
+            prev_dir=prev_dir,
+        )
+        jobs.append(displacement_calcs)
+
+        # Get DFT and harmonic forces
+        force_calcs = get_forces(
+            phonon_doc.force_constants,
+            phonon_supercell,
+            displacement_calcs.output,
+        )
+        jobs.append(force_calcs)
+
+        sigma_a_vals = {}
+
+        # Calculate atom-resolved sigma^A
+        if atom_resolved:
```

Comment:
Can this be merged into a single function that gets all sigma^A values? Right now it would setup a job for each one of these tasks that are all fairly quick